src/shared/watchdog.c

```diff
@@ -19,22 +19,6 @@ static char *watchdog_device;
 static usec_t watchdog_timeout;
 static usec_t watchdog_last_ping;
 
-static int enable_watchdog(void) {
-        int flags = WDIOS_ENABLECARD;
-
-        assert(watchdog_fd >= 0);
-
-        if (ioctl(watchdog_fd, WDIOC_SETOPTIONS, &flags) < 0) {
```

Comment:
honestly, even if you can drop this i wouldn't it's a nice way to confirm things are in order, and that we can issue ioctls and get sensible answers from the device, i.e. otherwise you can do `ln -s /dev/rtc0 /dev/watchdog0` and we'd just happily go on. I think some super basic checks after opening the device make sense.

i.e. sounds more like something where we should add a comment "Yeah, should be on anyway, we know, but let's make this explicit"

Reply:
When we open a device, we'll issue either WDIOC_GETTIMEOUT or WDIOC_SETOPTIONS so the case you describe can't happen unnoticed in practice.

So I still fail to see the point to issue a useless ioctl(), carry useless code. If the driver doesn't enable the device after opening it then it's a bug in the driver.

Reply:
Hmm, the errors from WDIOC_SETTIMEOUT are ignored though, and so are the errors from WDIOC_GETSUPPORT too

Reply:
Same when `enable_watchdog()` failed. All errors have been ignored so we leave the watchdog device opened and enabled but in an unknown state. We should probably close the device in these cases.

Reply:
I added a new commit that close the device in case of errors.